specs/governance/governor.md

```diff
@@ -0,0 +1,702 @@
+`GovernanceToken` contract.
+
+```solidity
+function getVotes(uint256 _account) external view returns (uint256);
+```
+
+#### `getVotesWithParams`
+
+Returns the voting power of an account at a specific timepoint given additional encoded parameters.
+
+```solidity
+function getVotesWithParams(address _account, uint256 _timepoint, bytes _params) external view returns (uint256);
+```
+
+#### `hasVoted`
+
+Returns whether `_account` has cast a vote on `_proposalId`.
+
+```solidity
+function hasVoted(uint256 _proposalId, address _account) external view returns (bool);
+```
+
+#### `hashProposal`
+
+Hashing function used to build a proposal id from proposal details.
+
+```solidity
+function hashProposal(address[] _targets, uint256[] _values, bytes[] _calldatas, bytes32 _descriptionHash) external view returns (uint256);
+```
+
+#### `hashProposalWithModule`
+
+Hashing function used to build a proposal id from proposal details and module address.
+
+```solidity
+function hashProposalWithModule(address _module, bytes _proposalData, bytes32 _descriptionHash) external view returns (uint256);
+```
+
+#### `manager`
+
+Returns the manager address of the `Governor` contract.
+
+```solidity
+function manager() external view returns (address);
+```
+
+#### `name`
+
+Returns the name of the `Governor` contract.
+
+```solidity
+function name() external view returns (string memory);
+```
+
+#### `proposalDeadline`
+
+Returns the timestamp for when voting on a proposal ends.
+
+```solidity
+function proposalDeadline(uint256 _proposalId) external view returns (uint256);
+```
+
+#### `proposalSnapshot`
+
+Returns the timestamp when the proposal snapshot was taken.
+
+```solidity
+function proposalSnapshot(uint256 _proposalId) external view returns (uint256);
+```
+
+#### `proposalThreshold`
+
+Returns the number of votes required in order for a voter to become a proposer.
+
+```solidity
+function proposalThreshold() external view returns (uint256);
+```
+
+#### `proposalVotes`
+
+Returns the distribution of votes for a proposal, ordered in `against`, `for`, and `abstain`.
+
+```solidity
+function proposalVotes(uint256 _proposalId) external view returns (uint256, uint256, uint256);
+```
+
+#### `quorum`
+
+Returns the quorum for a proposal, in terms of number of votes: `supply * numerator / denominator`.
+
+```solidity
+function quorum(uint256 _proposalId) external view returns (uint256);
+```
+
+#### `quorumDenominator`
+
+Returns the denominator for the quorum calculation.
+
+```solidity
+function quorumDenominator() external view returns (uint256);
+```
+
+#### `quorumNumerator`
+
+Returns the quorum numerator at a specific block number.
+
+```solidity
+function quorumNumerator(uint256 _blockNumber) external view returns (uint256);
+```
+
+#### `quorumNumerator` (block)
+
+Returns the quorum numerator at the current block number.
+
+```solidity
+function quorumNumerator() external view returns (uint256);
+```
+
+#### `state`
+
+Returns the state of a proposal.
```

Comment:
What are the possible states?

Reply:
specified the control flow in https://github.com/ethereum-optimism/specs/pull/315/commits/9a0fb5809772b4a6840fb165b1d4579658b6c7cb